packages/frontend/src/components/common/Filters/index.tsx

```diff
@@ -60,25 +60,28 @@ const FiltersForm: FC<Props> = ({
         (filterRules: FilterRule[]) => {
             const result = getFilterRulesByFieldType(fields, filterRules);
 
-            setFilters({
-                ...filters,
-                dimensions:
-                    result.dimensions.length > 0
-                        ? {
-                              id: uuidv4(),
-                              ...filters.dimensions,
-                              and: result.dimensions,
-                          }
-                        : undefined,
-                metrics:
-                    result.metrics.length > 0
-                        ? {
-                              id: uuidv4(),
-                              ...filters.metrics,
-                              and: result.metrics,
-                          }
-                        : undefined,
-            });
+            setFilters(
+                {
+                    ...filters,
+                    dimensions:
+                        result.dimensions.length > 0
+                            ? {
+                                  id: uuidv4(),
+                                  ...filters.dimensions,
+                                  and: result.dimensions,
+                              }
+                            : undefined,
+                    metrics:
+                        result.metrics.length > 0
+                            ? {
+                                  id: uuidv4(),
+                                  ...filters.metrics,
+                                  and: result.metrics,
+                              }
+                            : undefined,
+                },
+                true,
```

Comment:
I don't think we want to sync the state on update. This might be triggering multiple types while typing a string/number